ipywidgets/static/widgets/js/layout.js

```diff
@@ -104,7 +103,7 @@ define([
          */
         handleChange: function(trait, value) {
             this.displayed.then(_.bind(function(parent) {
-                if (parent && value) {
```

Comment:
If you make this change, and make sure the traits default to `None`, I think you don't have to remove the shorthand trait